### PR TITLE
Implement bitcoind v23 rpc 

### LINKED ISSUE #4292: Implement bitcoind v23 rpc
https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-April/020321.html

### PR DESCRIPTION
**Pull request by solo1g**
Closes #4292 

v23 release notes for reference: https://lists.linuxfoundation.org/pipermail/bitcoin-core-dev/2022-April/000111.html

### CODE REVIEW

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala [hunk 1]
```diff
@@ -350,10 +352,10 @@ sealed trait BitcoindVersion
 object BitcoindVersion extends StringFactory[BitcoindVersion] {
 
   /** The newest version of `bitcoind` we support */
-  val newest: BitcoindVersion = V22
+  val newest: BitcoindVersion = V23
```
Comment (Christewart) [inline bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala:355]: Can you try reverting this and see if the `lndTest` issues go away? It is probably worth looking at the lnd release notes to see if they fully support `v23` of bitcoind

Comment (solo1g) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala:355]: Made no change and just ran the tests again and they pass this time :man_shrugging:.

Comment (Christewart) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala:355]: Hmm @benthecarman any educated guesses from you? We haven't been merging a ton of PRs lately, but this does seem like more failures than normal. Ben recently merged some work to fix one of the consistently failing test cases in #4361

Comment (benthecarman) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala:355]: I have been running lnd v0.14.3 with bitcoind v23 with no issues. Looks like CI is passing now

**Comment by solo1g** [comment]
Squashed the rerun commits. Seems like the issue got fixed with the recent changes.

File: bitcoind-rpc/bitcoind-rpc.sbt [hunk 2]
```diff
@@ -38,14 +39,20 @@ TaskKeys.downloadBitcoind := {
   logger.debug(
     s"(Maybe) downloading Bitcoin Core binaries for versions: ${versions.mkString(",")}")
 
-  val (platform, suffix) =
+  def getPlatformAndSuffix(version: String): (String, String) = {
     if (Properties.isLinux) ("x86_64-linux-gnu", "tar.gz")
-    else if (Properties.isMac) ("osx64", "tar.gz")
+    else if (Properties.isMac)
+      version match {
+        case "23.0" => ("x86_64-apple-darwin", "tar.gz")
```
Comment (Christewart) [inline bitcoind-rpc/bitcoind-rpc.sbt:46]: :tada:

**Review by Christewart — APPROVED** [review]
Really nice job @shreyanshyad :tada:

**Pull request merged**